packages/playwright-test/src/reporters/junit.ts

```diff
@@ -133,6 +133,94 @@ class JUnitReporter implements Reporter {
     };
     entries.push(entry);
 
+    // Xray Test Management supports testcase level properties, where additional metadata may be provided
+    // some annotations are encoded as value attributes, other as cdata content
+    const properties: XMLEntry = {
+      name: 'properties',
+      children: [] as XMLEntry[]
+    };
+
+    const textValueBasedAnnotations = ['test_id', 'test_key', 'test_summary', 'requirements'];
+    for (const annotationType of textValueBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType,
+            value: (annotation?.description ? annotation.description : '')
+          }
+        };
+        properties.children?.push(property);
+      }
+    }
+
+    const textContentBasedAnnotations = ['test_description'];
+    for (const annotationType of textContentBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType
+          },
+          text: annotation.description
+        };
+        properties.children?.push(property);
+      }
+    }
+
+    // attachments are optionally embed as base64 encoded content on inner elements
+    const embedAttachments = test.annotations.find(annotation => {
+      return annotation.type === 'embed_attachments_in_report';
+    });
+    if (embedAttachments?.description === 'true') {
+      const evidence: XMLEntry = {
+        name: 'property',
+        attributes: {
+          name: 'testrun_evidence'
```

Comment:
Let's add an option to the junit reporter: `{ embedAttachmentsAsProperty: 'testrun_evidence' }`? This seems a very specific name to put here.

Reply:
That `testrun_evidence` name is the one that Xray parses but I'm ok having it configurable, just let me know if you want to proceed that way.
Just for information, JUnit doesn't specify how to embed attachments in the report. This is something specified by the Xray team (even though I've validated it with JUnit5 team leader at high-level just to know if it would hurt the "schema" or go against some principles), that eventually other tools can take advantage of :)

Reply:
Let's go with a single global option for the junit reporter. See my other comment with suggested options.